awx/api/views/analytics.py

```diff
@@ -0,0 +1,274 @@
+
+    @staticmethod
+    def _forward_response(response):
+        try:
+            content_type = response.headers['content-type'] or ''
+            if content_type.find('application/json') != -1:
+                return Response(response.json(), status=response.status_code)
+        except Exception as e:
+            logger.error(f"Analytics API: Response error: {e}")
+
+        return Response(response.content, status=response.status_code)
+
+    def _send_to_analytics(self, request, method):
+        try:
+            headers = self._request_headers(request)
+
+            self._check_upload_enabled()
+            url = self._get_analytics_url(request.path)
+            rh_user = self._get_rh_user()
+            rh_password = self._get_rh_password()
+
+            if method == "GET":
+                response = requests.get(url, auth=(rh_user, rh_password), verify=CERT_PATH, params=request.query_params, headers=headers, json=request.data)
+            elif method == "POST":
+                response = requests.post(url, auth=(rh_user, rh_password), verify=CERT_PATH, params=request.query_params, headers=headers, json=request.data)
+            else:
+                return self._error_response(ERROR_UNSUPPORTED_METHOD, method, remote=False, status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+            #
+            # Missing or wrong user/pass
+            #
+            if response.status_code == status.HTTP_401_UNAUTHORIZED:
+                text = (response.text or '').rstrip("\n")
+                return self._error_response(ERROR_UNAUTHORIZED, text, remote=True, remote_status_code=response.status_code)
+            #
+            # Not found, No entitlement or No data in Analytics
+            #
+            elif response.status_code == status.HTTP_404_NOT_FOUND:
+                return self._error_response_404(response)
+            #
+            # Success or not a 401/404 errors are just forwarded
+            #
+            else:
+                return self._forward_response(response)
+
+        except MissingSettings as e:
+            return self._error_response(e.args[0], remote=False)
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Analytics API: Request error: {e}")
+            return self._error_response(ERROR_UNKNOWN, str(e), remote=False, status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
+        except Exception as e:
+            logger.error(f"Analytics API: Error: {e}")
+            return self._error_response(ERROR_UNKNOWN, str(e), remote=False, status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+class AnalyticsGenericListView(AnalyticsGenericView):
+    def get(self, request, format=None):
+        return self._send_to_analytics(request, method="GET")
+
+    def post(self, request, format=None):
+        return self._send_to_analytics(request, method="POST")
+
+
+class AnalyticsGenericDetailView(AnalyticsGenericView):
```

Comment:
This class is only used in one place. Are we thinking that we might have additional endpoints someday and that is why its separated out?

Reply:
yes, I expect there could be more `/<path>/<slug>` endpoints